React Quill Image Upload

rbennettcw opened the pull request:
Implements image upload for the React Quill component.

## Link to Issue
Closes: #2932

## Description of Changes
- Adds drag and drop image uploads for the react quill component

## Test Plan
- For any react quill component, drag an image to it, wait for the upload to succeed, confirm the image shows in the position it was dropped. Click save. Confirm the image shows correctly in the read-only view.

## Deployment Plan
N/A

## Other Considerations
N/A

masvelio reviewed (commented):
<img width="1145" alt="CleanShot 2023-03-21 at 09 17 03@2x" src="https://user-images.githubusercontent.com/14819225/226675637-7750b649-a4a2-4841-82dd-a09bdc243580.png">
 
Does not work for me in thread. Works in comments. It should work in every place in the app where we use quill. 

Also, on prod, there is a loader visible when I am uploading the image. It is missing here

rbennettcw commented:
Tested image upload for the following features:
- Create comment (OK)
- Edit comment (OK)
- Edit profile (OK, but saves as text only)
- New thread (OK)
- Edit thread (OK)
- Edit topic modal (OK)
- New topic modal (OK)

rbennettcw commented:
Alright, this should be good to go now. Fixed erroneous merge resolutions, replaced JQuery with Axios, added loading indicator, tested the component in multiple features.

masvelio commented on packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:
```diff
@@ -22,19 +43,87 @@ const ReactQuillEditor = ({
+        value={contentDelta}
+        onChange={handleChange}
+        modules={{
+          toolbar: ([[{ header: 1 }, { header: 2 }]] as any).concat([
```
cool stuff!

masvelio reviewed (approved):
looks good and works fine for me! The only thing that is missing is the code formatting. Please always use prettier on every file you work on

jnaviask commented:
Getting really small comment windows
![2023-03-22-114921_1038x886_scrot](https://user-images.githubusercontent.com/1860629/226961859-e273375c-1d92-42a3-b469-ca51ac3b7799.png)

Is this an intentional change?

Beyond that, seems to work great!

rbennettcw commented:
> looks good and works fine for me! The only thing that is missing is the code formatting. Please always use prettier on every file you work on

Thanks, I've setup my local config to apply formatting/linting. Will apply to future PRs if that's alright.

jnaviask commented:
One other thing I noticed is that images aren't counted as valid text, so users can't make an image-only post or reply. Otherwise, works great & can merge.

rbennettcw commented:
@jnaviask I've pushed a fix to acknowledge image as a valid input and allow image-only posts.

jnaviask commented:
Amazing stuff!
